Remotion/ObjectBinding/Core/BusinessObjectProvider.cs

```diff
@@ -174,7 +175,13 @@ public IBusinessObjectService GetService (Type serviceType)
 
       // Optimized for memory allocations
       if (_getServiceStoreValueFactory == null)
-        _getServiceStoreValueFactory = type => _serviceFactory.CreateService (this, type);
+      {
+        _getServiceStoreValueFactory = type =>
+            Assertion.IsNotNull (
```

Comment:
Docs state `Must not be <see langword="null" />`, is this outdated?

Reply:
wonderful. bug either in doc or in impl. need to check the usgages and implications of this returning null

Reply:
All usages apart from one do have a nullcheck on the callsite, the one check must be added then.

Reply:
okay. For the location of the nullcheck: I think it may be better to add the nullguard outside of the cache to allow the exception to be thrown each time the API is used instead of repeatedly failing during the cache-add. Yes, we cache the null result then.